### DISCUSSION TITLE
UVC class support for USB3

### DISCUSSION THREAD

**Discussion by sbaratam** [hathach/tinyusb#2800] [Q&A]
Hi,
I am looking at using tinyUSB for use-case with UVC class. I can see video capture application and UVC class implementation in tinyUSB, but in readme it is mentioned as "UVC 1.5: work in-progress". Can i use tinyUSB to verify video class for USB3? Should i expect any specific issues with it?

**Comment by josuah** [comment]
UVC is generally independent of the underlying USB version: the standard was published for USB2 HighSpeed as fastest speed, and this is the same standard that USB3 SuperSpeed cameras uses.

The larger challenge will be to find an USB3 hardware and enabling USB3 handling on tinyUSB.
We are working on something like this for Zephyr, and I would be happy to share insights and help debugging wrt tinyUSB implementation if someone wanted to try porting tinyUSB to an USB3-capable hardware.

The other solution is Linux gadget mode: it has drivers for USB3 controllers and an UVC implementation for device mode.
This is what the "turn your Android phone into a webcam" application uses AFAIR.

Though, if you already have the USB3 controller, adding UVC to it should be relatively straightforward! :)
